react/react_classes/README.md

```diff
@@ -0,0 +1,113 @@
+
+Our Animal prototype includes the ability for different instances to have different names - just as two dogs might share many qualities but be named differently. We add `name` as an argument to the constructor, and then we immediately apply it as an attribute to the class by placing it on `this`. 
+
+`this` is a keyword in JavaScript that means different things depending on context. Generally, it refers to the greater context in which it was placed - in this case, the class object itself. By writing `this.name = name`, we place the `name` we passed into the constructor's argument on the `this` object, which is, in this case, the instance of `Animal` we're constructing.
+
+That's a lot going on under the hood, but in practice, the notation for creating an instance of `Animal` is fairly straightforward:
+
+```js
+const myDog = new Animal('Luna')
+
+myDog.name
+// => 'Luna'
+
+myDog.speak('woof')
+// => 'Luna says "woof"'
+```
+
+# Extending classes
+
+Another feature available with classes is the `extends` keyword. This allows us to define a type of object with the same methods (and properties) of an existing type of object, plus whatever new methods and properties we want to define. For example, we may define an object of type `Dog` that extends the `Animal` class:
+
+```js
+class Dog extends Animal {
+  constructor(name, breed){
+    super(name)
+    this.breed = breed;
+  }
+
+  bark() {
+    return `${this.name} barks.`
+  }
+}
+
+let myDog = new Dog('Luna')
+
+myDog.speak('haha!')
+// => 'Luna says "haha!"'
+
+myDog.bark()
+// => 'Luna barks.'
+```
+
+In this case, the method `speak` is a method on the `Animal` class, but the method `bark` is unique to the `Dog` class. We can make Dogs `speak` (as they are animals), but we cannot make any Animal `bark` (as they are not, necessarily, dogs). In other words, all forks are utensils, but not all utensils are forks.
+
+When extending a class, in order to make sure the parent class gets the data it needs, we need to call `super()` with the arguments the parent class constructor was expecting. Above, the `super` function calls the constructor for the `Animal` class with the provided name.
+
+# Class components in React
+
+Now that we understand the basics of how classes use `extends` to inherit functionality, we get a better impression of why classes might have utility in React. Consider the following functional React component:
+
+```jsx
+import React from 'react';
+
+const Example = () => {
+  return <h1> Hello World! </h1>;
+}
+```
+
+And its class-based equivalent:
+
+```jsx
+import React from 'react';
+
+class Example extends React.Component {
+  constructor() {
+    super()
+  }
+
+  render() {
+    return <h1> Hello World! </h1>;
+  }
+}
```

Comment:
Question on how much React the fellows know at this point and the position of this lecture. Is this their first exposure to seeing a react functional and class component?

Reply:
@coreyladovsky This will be lesson 2. In lesson 1 they saw functional components only. So this is their first exposure to seeing a class component, and the second time seeing functional components.

The next lessons after this are props and state. They'll need to know about class components before the state lesson. I think it would be pretty painless to do this order if you think it will help. We'll just need to be sure not to use class components in the props lesson.
1. Intro
2. Props
3. Class components
4. State

Reply:
For what it's worth, even if it doesn't have immediate utility, I think the earlier we can be teaching class syntax, the better.

Reply:
Okay let's leave the order as original planned then:
1. Intro
2. Class components
3. Props
4. State